nrclient/nrclient.go

```diff
@@ -0,0 +1,67 @@
+package nrclient
+
+import (
+	"bytes"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"log"
+	"net/http"
+	"time"
+
+	"github.com/newrelic-fluent-bit-output/config"
+)
+
+type NRClient struct {
+	client *http.Client
+	config config.NRClientConfig
+}
+
+func NewNRClient(cfg config.NRClientConfig, proxyCfg config.ProxyConfig) (*NRClient, error) {
+	httpTransport, err := buildHttpTransport(proxyCfg, cfg.Endpoint)
+	if err != nil {
+		return nil, fmt.Errorf("building HTTP transport: %v", err)
+	}
+
+	nrClient := &NRClient{
+		client: &http.Client{
+			Transport: httpTransport,
+			Timeout:   5 * time.Second,
```

Comment:
should this be configurable?

Reply:
Discussed above: let's agree if we want to introduce any more configurability, or if we can just keep the fixed values for this PR and parametrize them later.